Services/Tracking/classes/class.ilTrQuery.php

```diff
@@ -1087,6 +1093,27 @@ public static function getParticipantsForObject(int $a_ref_id): ?array
         );
     }
 
+    /**
+     * @param int[] $user_ids
+     * @return int[]
+     */
+    protected static function filterOutUsersWithoutData(array $user_ids): array
+    {
+        $res = [];
+        foreach ($user_ids as $user_id) {
```

Comment:
To avoid this iteration - under normal circumstances - you could call userExists() with all user_ids and iterate only in case the result is different to the number of users.

Reply:
done, thanks!